### PR TITLE
fix: vite include optmize

### PR DESCRIPTION
**Pull request by himself65**
force vite to pre-bundle such packages to avoid dual instances in frontend

https://vitejs.dev/guide/dep-pre-bundling.html

### CODE REVIEW

File: packages/waku/src/lib/middleware/rsc/ssr.ts [hunk 1]
```diff
@@ -328,13 +342,18 @@ const rectifyHtml = () => {
...
-): Promise<readonly [ReadableStream, Context] | null> => {
+export type SsrHtmlContext<UserContext> = {
+  htmlStr: string; // Hope stream works, but it'd be too tricky
+  command: 'dev' | 'build' | 'start';
+  config: ResolvedConfig;
+  ssrConfig: SsrConfig;
+  userContext: UserContext | undefined;
+};
+
+export const ssrHtml = async <UserContext>(
```
Comment (himself65) [inline packages/waku/src/lib/middleware/rsc/ssr.ts:353]: I think rename it to `ssr` will make more sense

**Comment by himself65** [comment]
```shell
41765 stdout:  ready: Listening on http://localhost:41765/
37997 stderr:  WebSocket server error: Port is already in use
37997 stderr:  5:18:43 AM [vite] Pre-transform error: Failed to load url /assets/main-Z4dkQC16.js (resolved id: /assets/main-Z4dkQC16.js). Does the file exist?
37997 stderr:  5:18:43 AM [vite] warning: 
/index.html
1  |  
2  |  globalThis.__waku_module_cache__ = new Map();
3  |  globalThis.__webpack_chunk_load__ = (id) => import(id).then((m) => globalThis.__waku_module_cache__.set(id, m));
   |                                                     ^^
4  |  globalThis.__webpack_require__ = (id) => globalThis.__waku_module_cache__.get(id);
5  |  if (import.meta.hot && !globalThis.__WAKU_HMR_CONFIGURED__) {
The above dynamic import cannot be analyzed by Vite.
See https://github.com/rollup/plugins/tree/master/packages/dynamic-import-vars#limitations for supported dynamic import formats. If this is intended to be left as-is, you can use the /* @vite-ignore */ comment inside the import() call to suppress this warning.

  Plugin: vite:import-analysis
  File: /index.html?html-proxy&index=0.js
37997 stderr:  (node:11800) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
37997 stdout:  Cannot render RSC Error: cannot create worker with different command
    at getWorker (file:///home/runner/work/waku/waku/packages/waku/dist/lib/middleware/rsc/worker-api.js:8:19)
    at renderRSC (file:///home/runner/work/waku/waku/packages/waku/dist/lib/middleware/rsc/worker-api.js:67:20)
    at ssrHtml (file:///home/runner/work/waku/waku/packages/waku/dist/lib/middleware/rsc/ssr.js:228:35)
    at async file:///home/runner/work/waku/waku/packages/waku/dist/lib/middleware/rsc.js:135:36
error: Failed to load resource: the server responded with a status of 500 (Internal Server Error)
×.
```

File: packages/waku/src/lib/middleware/rsc.ts [hunk 2]
```diff
@@ -183,31 +205,15 @@ export function rsc<
...
-          item.file === fname &&
-          item.url !== pathStr &&
-          !item.url.includes('?html-proxy')
-        ) {
-          res.setHeader('Content-Type', 'application/javascript');
-          res.setStatus(200);
-          endStream(res.stream, `export * from "${item.url}";`);
-          return;
-        }
-      }
```
[the remark this answers is not in the document]
Comment (dai-shi) [reply packages/waku/src/lib/middleware/rsc.ts:210]: yeah, probably. Please try removing it.

**Review by himself65 — COMMENTED** [review]
I guess things here is:
`react-dom/client -> react-dom -> INTERNAL_XXX`
`waku -> react-dom -> INTERNAL_XXX`

since `waku` is excluded, somehow vite will bundle react-dom twice

**Comment by dai-shi** [comment]
If this is only about 11_form example, the reason was one of client components depends on `react-dom`.

File: packages/waku/src/lib/handlers/handler-dev.ts [hunk 3]
```diff
@@ -45,7 +45,7 @@ export function createHandler<
     const mergedViteConfig = await mergeUserViteConfig({
       base: config.basePath,
       optimizeDeps: {
-        include: ['react-server-dom-webpack/client'],
+        include: ['react-server-dom-webpack/client', 'react', 'react-dom'],
```
Comment (dai-shi) [inline packages/waku/src/lib/handlers/handler-dev.ts:48]: In fact, I used add them, but removed afterwards as they seemed unnecessary, which is wrong.

**Review by dai-shi — APPROVED** [review]
Looks good!

**Pull request merged**